src/service/video-manager-impl.js

```diff
@@ -444,6 +444,12 @@ class VideoEntry {
     listen(video.element, VideoEvents.UNMUTED, () => this.muted_ = false);
     listen(video.element, VideoEvents.ENDED, () => this.videoEnded_());
 
+    listen(video.element, VideoAnalyticsEvents.CUSTOM, e => {
+      const data = getData(e);
```

Comment:
We would want to allow them to pass some arbitrary data in addition to a string name. Ideally that data becomes available under the existing https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/amp-video-analytics.md#common-variables which every video event exposes. 
(maybe namespaces under a `customData` property BUT we need to make sure `VIDEO_STATE(selector,property)` can access it (e.g. `VIDEO_STATE(selector,'customData.foo')` works)

Reply:
I don't think the nested property like `customData.foo` would work with the current implementation. The data for the `VIDEO_STATE` var sub comes from `VideoEntry#getAnalyticsDetails`

https://github.com/ampproject/amphtml/blob/f1159414bfb0a0dfafb6e322ce7c9611896f3e72/src/service/video-manager-impl.js#L866-L877

With the current implementation, you could expose the custom analytics data as a stringified map in that function, like `getPlayedRangesJson`. But to get the sub-properties you'd need to implement the dot-notation access.

Reply:
we could also flatten and merge them with the existing one. Maybe prefix with `customData_foo`, `customData_bar`, etc.. to avoid conflicts

Reply:
Done.